src/topic_based_system.cpp

```diff
@@ -191,7 +231,15 @@ hardware_interface::return_type TopicBasedSystem::read(const rclcpp::Time& /*tim
     if (it != joints.end())
     {
       auto j = static_cast<std::size_t>(std::distance(joints.begin(), it));
-      joint_states_[POSITION_INTERFACE_INDEX][j] = latest_joint_state_.position[i];
+      if (sum_wrapped_joint_states_)
+      {
+        sumRotationFromMinus2PiTo2Pi(latest_joint_state_.position[i], previous_joint_state_.position[i],
```

Comment:
Do we really need the new variable `previous_joint_state_`? `joint_states_[POSITION_INTERFACE_INDEX][j]` is already holding the previous values, we could rewrite the code as `joint_states_[POSITION_INTERFACE_INDEX][j] = sumRotationFromMinus2PiTo2Pi(latest_joint_state_.position[i], joint_states_[POSITION_INTERFACE_INDEX][j])`

Reply:
I was being lazy and didn't want to unwrap the summed rotation but the angles library does this for me so :+1: